internal/backend/db/dbgorm/dbgorm.go

```diff
@@ -50,51 +45,59 @@ func New(z *zap.Logger, cfg *Config) (db.DB, error) {
 		return nil, err
 	}
 
-	db := &gormdb{z: z, cfg: cfg}
+	return &gormdb{z: z, cfg: cfg}, nil
+}
+
+func (db *gormdb) GormDB() (r, w *gorm.DB) { return db.rdb, db.wdb }
+
+func connect(_ context.Context, z *zap.Logger, cfg *Config) (r *gorm.DB, w *gorm.DB, err error) {
+	gormCfgFn := func(cfg *gorm.Config) { cfg.SkipDefaultTransaction = true }
 
-	if cfg.Type == "sqlite" {
-		db.mu = new(sync.Mutex)
+	r, err = gormkitteh.OpenZ(z, cfg, gormCfgFn)
+	if err != nil {
+		err = fmt.Errorf("opendb: %w", err)
+		return
 	}
 
-	return db, nil
-}
+	var sqlDB *sql.DB
+	if sqlDB, err = r.DB(); err != nil {
+		return
+	}
 
-func (db *gormdb) GormDB() *gorm.DB { return db.db }
+	n := cfg.MaxOpenConns
+	if n == 0 {
+		n = max(4, runtime.NumCPU())
+	}
 
-func connect(_ context.Context, z *zap.Logger, cfg *Config) (*gorm.DB, error) {
-	client, err := gormkitteh.OpenZ(z, cfg, func(cfg *gorm.Config) {
-		cfg.SkipDefaultTransaction = true
-	})
-	if err != nil {
-		return nil, fmt.Errorf("opendb: %w", err)
+	sqlDB.SetMaxOpenConns(n)
+
+	// in memory db doesn't need any separation between writing and reading.
+	if cfg.Type != "sqlite" || strings.HasPrefix(cfg.DSN, ":memory:") {
```

Comment:
I prefer to have utility function as as `IsSQLite` or `DBType`. There's code in `Setup` that only checks `db.cfg.Type`.

Reply:
this also checks the DSN, not just the db type.